backend/sftp/file.go

```diff
@@ -252,7 +252,7 @@ func (f *File) Seek(offset int64, whence int) (int64, error) {
 // Write calls the underlying sftp.File Write.
 func (f *File) Write(data []byte) (res int, err error) {
 
-	sftpfile, err := f.openFile(os.O_RDWR | os.O_CREATE)
+	sftpfile, err := f.openFile(os.O_WRONLY | os.O_CREATE)
```

Comment:
I think this can work but this won't address the same issue if we try to seek. Seek is the tricky one because it could be for both read or write.  The way we were approaching that with the append vs overwrite issue is doing a remote-to-local copy before the first write, then acting on the temp, until close write back to remote.  Here we're doing direct remote writes/seek/etc.